models/netsuite2/intermediate/base/int_netsuite2__transaction_lines.sql

```diff
@@ -28,6 +54,11 @@ joined as (
     left join transaction_accounting_lines
         on transaction_lines.transaction_line_id = transaction_accounting_lines.transaction_line_id
         and transaction_lines.transaction_id = transaction_accounting_lines.transaction_id
+    left join accounting_books
+        on accounting_books.base_book_id = transaction_accounting_lines.accounting_book_id
+    where accounting_books.base_book_id is not null
```

Comment:
Would you be able to elaborate on this where clause addition? Would it be relevant to still include records that are `null` for a base book id?

Reply:
Definitely. NetSuite has two configurations for multibook: 
1) True multibook - transactions are fully replicated and present in the transactional data 
2) Adjustment only books - adjustments are stored in the database, but reporting is presented as base book + adjustments.

In both instances consolidated exchange rates are per book.

In the case of adjustment only books, the admin defines a base book during setup. When the `base_book_id` is not null that means the base book transactions should all be replicated onto the adjustment only book for reporting. We would not want to include records where the `base_book_id` is null.